### ISSUE TITLE
[BUG] Quick Resume fails

### ISSUE THREAD

**Issue by mtaki14** [nixgates/plugin.video.seren#373]
**Describe the bug**

Right click > Quick Resume Show results in "Seren Error Check the log for more information"

The same error appears if I enable the "Force Auto Episode Resume" option.

**Expected behavior**
For Seren to pick up from the episode where it left off or start the following episode (depending on if the last watched episode was watched in its entirety)

**Kodi Version (please complete the following information):**
 - Kodi Version 18.9.0
 - Seren Version 2.0.10

[kodi.log](https://github.com/nixgates/plugin.video.seren/files/5781982/kodi.log)

**Comment by SerpentDrago** [comment]
If you have resume show enabled in seren and just click on a show like normal , does it resume ? 

or is it just the context menu > quick resume that just doesn't work , yet resuming still actually works ?

**Comment by mtaki14** [comment]
Neither works. Some more background, I only started using trakt with seren after version 2 was released.

**Comment by SerpentDrago** [comment]
by default seren should just resume . it doesn't requre you to use the context menu . 

make sure your trakt account is authed , and that seren has finished syncing .

**Comment by mtaki14** [comment]
Trakt is authed and works otherwise

**Comment by SerpentDrago** [comment]
https://i.imgur.com/Cx2uYL2.png

insure that is enabled

**Comment by mtaki14** [comment]
![seren_playback_settings](https://user-images.githubusercontent.com/7987430/103902587-6a978a00-5103-11eb-9430-fff835bb34ec.png)

p/s
I assume the log file in the original post would be of some help

**Comment by SerpentDrago** [comment]
ohh boy you have some OLD old addons and stuff installed 

 plugin.video.exodusredux v2.0.3a installed
script.module.exodusredux v2.0.5 installed
script.module.lambdascrapers v2.0.0 installed



just some of them , i suggest you go though uninstall any old repos and addons you don't use , then go though your kodi addon dependency section and remove any orphaned entries . 

then restart , make sure all your addons are updated . Then reauth Trakt in seren , and then rebuild trakt db from its settings , let it complete (it will take a bit )

**Comment by mtaki14** [comment]
Dully noted.

This happens on another system as well that only has Fen installed alongside Seren so this most likely has something to do with the way Seren interacts with my Trakt account.

**Comment by SerpentDrago** [comment]
is this just Episodes or movies also ?   does it ever prompt you to resume (popup that you want t resume or not ) 

i just tested on a movie , on windows 10 with latest seren , and it asked even though i had it on automatic , so at the very least that is broken . but it did resume correctly

**Comment by mtaki14** [comment]
Can't recall movies but for individual episodes it does ask me to resume from where it left off with a popup.

**Comment by SerpentDrago** [comment]
And just tested on a episode , and it prompted and resumed when clicked .

**Comment by SerpentDrago** [comment]
such as this notice the time stamp 

https://i.imgur.com/akvpKGu.png

**Comment by mtaki14** [comment]
I can't even get into the season/ episode selection screens with auto resume enabled.

This is with the defaul skin

**Comment by SerpentDrago** [comment]
OHHHH , one sec , change this > 

![image](https://user-images.githubusercontent.com/21133858/103903963-9fd3b600-50ca-11eb-8a2b-e23d874bca1b.png)


UNCHECK force auto episode resume , this does NOT do what you think it does , it makes it so when you click a SHOW , it auto trys to resume the last EPISODE you played . IMHO it should be disasbled by default and causes nothing but proplems

@nixgates  could we please not have this enabled by default , i have had nothing but support and issues with this setting ...

**Comment by mtaki14** [comment]
That doesn't change anything, I already tried

**Comment by SerpentDrago** [comment]
if your not getting to seasons , episodes , THAT setting is what controls that , TRust me .

what happens when you context menu a show and then "browse show / browse season " 

if clicking a SHOW , just starts playback instead of bringing you to the seasons and then episodes , then this setting is absolutely the issue

**Comment by mtaki14** [comment]
I meant it doesn't change the fact that auto resume still doesn't work. 

It takes me to the season select screen instead of resuming (when auto resume is enabled)

**Comment by SerpentDrago** [comment]
good ok , then browse to the corrrect season , then the episode ,  and see if it resumes properly that episode . 

you need to be clear when saying it doesn't change anything . The whole thing i'm testing is that if you disable auto show resume . and actually browse to the correct episode , if it will actually resume that episode .

**Comment by mtaki14** [comment]
Just to be clear. Is auto resume supposed to continue playback of the next episode of where you left off or does it only work if you left an episode in the middle?

**Comment by SerpentDrago** [comment]
![image](https://user-images.githubusercontent.com/21133858/103904750-b2022400-50cb-11eb-9bd8-9752905c8258.png)


that is so when you click a SHOW , it instead of showing you seasons / episodes , it just starts playing back the next un watched episode in that series . 


![image](https://user-images.githubusercontent.com/21133858/103904833-d1994c80-50cb-11eb-8e1c-6f905f664d64.png)


That allows you to Resume watching something (say your in the middle of it and back out , next time you play it , it will ask to resume to the last place you were at )_

**Comment by SerpentDrago** [comment]
@nixgates  This is a confusing mess lol

**Comment by mtaki14** [comment]
So in that auto force resume is exactly what I thought it was and is what is not working properly

**Comment by mtaki14** [comment]
So auto force resume and context menu > quick resume invoke the same function, correct?

**Comment by SerpentDrago** [comment]
agian nothing is called auto force resume

do you mean  "force auto episode resume"  it is very important you be clear .

again this setting should do nothing more then just auto scrape / start playback of the next unwatched episode . 



YES , "force auto episode resume"  IS quickresume , and quick resumes to the latest unwatched episode , aka netflix like function

**Comment by mtaki14** [comment]
> do you mean "force auto episode resume" it is very important you be clear .

Yes, sorry I meant that

**Comment by SerpentDrago** [comment]
the question is , is it just the context menu that doesn't work , or does the actual function if set so in setting not work , 

aka what happens when you click a show (not a episode , not a movie ) and SHOW , and have "force auto episode resume" enabled .

**Comment by mtaki14** [comment]
Same error as mentioned in the op

**Comment by SerpentDrago** [comment]
Thank you :) haha 

and just to be clear if you disable "force auto episode resume"  everything works per typical addons , (you see seasons , then episodes ) and you can click a previous stoped episode and it resumes ? 


sorry we were both prob confusing

**Comment by mtaki14** [comment]
Yes

**Comment by SerpentDrago** [comment]
@nixgates 

TLDR 

Both Context menu "quick resume "  and seren's netflix like function  " "force auto episode resume" "  are broken and cause errors

**Comment by mtaki14** [comment]
To be fair, the original report was refering to the quick resume functionality specifically until you brought regular resuming into the conversation.

But on the bright side now I have no qualms about what each function does lol

**Comment by SerpentDrago** [comment]
Yeh i understand that .. it was the way you worded it 

Expected behavior
For Seren to pick up from the episode where it left off.

pickup from where it left off implies you Resuming a bookmark , time stamp in the episode 

at least how i read it 

like i said , i prob made it more confusing , sorry 

but the entire point was to see if the function worked from outside the context menu . and it was not just the context menu but the actual full function

**Comment by SerpentDrago** [comment]
Long story short , i need more coffee :) have a good day , sorry nix

**Comment by mtaki14** [comment]
> Expected behavior
For Seren to pick up from the episode where it left off.

Yeah, in retrospect this could have been worded in a much better way

**Comment by mtaki14** [comment]
I edited the original description to make it clearer

**Comment by nixgates** [comment]
@mtaki14 ah sorry for my delay, thankfully this is already fixed for the next update! Was identified just the other day by testers :)

**Comment by mtaki14** [comment]
Thanks, I'll report back after the next update

**Comment by JimiJoe** [comment]
Im having the same issues with a fairly new installation of kodi (approximately 4 months). The netflix style worked great until approximately 1 or 2 weeks ago. Restore full kodi backup did not solve it. Is it helpful if I post my log? Not sure how I can add the log to this post as an attahment though, should I just copy/paste the log in full text?

**Comment by mtaki14** [comment]
@JimiJoe nixgates said that it should be fixed in the next release so I think better wait for that first.

**Comment by JimiJoe** [comment]
Ok. Kodi log by the way also showed it might have somethingbto do with nixgates skin helper service.

**Comment by mtaki14** [comment]
@nixgates It works now but it takes me to play the last episode that is already marked as watched in Trakt instead of the following one.

**Comment by nixgates** [comment]
@mtaki14 ah bother, ok gove me a bit. I'm working on yet another update right now. Will scoop this up as well.

**Comment by mtaki14** [comment]
I'll watch the following episode, re-enable the option and check again if it now picks up the right one.

**Comment by mtaki14** [comment]
Seems to be working now, after the fact

**Comment by nixgates** [comment]
@mtaki14 good to see! I'll close this off :)
